src/Components/Login/login.js

```diff
@@ -47,31 +48,54 @@ export default function Login() {
         return email.length > 0 && password.length > 0;
     }
 
-    function handleSubmit(event) {
+    async function handleSubmit(event) {
         event.preventDefault();
         const errors = {};
 
-        auth.signInWithEmailAndPassword(email, password).catch((error) => {
-            errors.login = "Email or password is incorrect.";
-            const errorCode = error.code;
+        auth.setPersistence(Auth.Persistence.NONE);
 
-            switch (errorCode) {
-                case "auth/invalid-email":
-                    errors.email = "Please enter a valid email.";
-                    break;
-                case "auth/user-disabled":
-                    errors.email = "Email has been disabled.";
-                    break;
-                case "auth/wrong-password":
-                    errors.login = "Email or password is incorrect.";
-                    break;
-                default:
-                    errors.login = "Email or password is incorrect.";
-                    break;
-            }
-            setAllErrors(errors);
-        });
+        await auth
+            .signInWithEmailAndPassword(email, password)
+            .then(async () => {
+                // Get the user's ID token as it is needed to exchange for a session cookie.
+                await auth.currentUser.getIdToken().then(async (idToken) => {
+                    try {
+                        const response = await httpPost(
+                            "/admin_login",
+                            {
+                                idToken: idToken,
+                            },
+                            true
+                        );
 
+                        return response;
+                    } catch (error) {
+                        alert(error.response.data.message);
+                        errors.login = error.response.data.message;
+                    }
+                });
+            })
+            .catch((error) => {
+                errors.login = "Email or password is incorrect.";
+                const errorCode = error.code;
+
+                switch (errorCode) {
+                    case "auth/invalid-email":
+                        errors.email = "Please enter a valid email.";
+                        break;
+                    case "auth/user-disabled":
+                        errors.email = "Email has been disabled.";
+                        break;
+                    case "auth/wrong-password":
+                        errors.login = "Email or password is incorrect.";
+                        break;
+                    default:
+                        errors.login = "Email or password is incorrect.";
+                        break;
+                }
```

Comment:
I'm not clear what `errors` is.  Would it make sense to use an enum to store the error messages so it's easier to change?

Reply:
Will clean up. This was consistent with the work here before, but I agree it can be done better.